sphinx/ext/autosummary/__init__.py

```diff
@@ -780,6 +782,8 @@ def setup(app):
     app.connect('doctree-read', process_autosummary_toc)
     app.connect('builder-inited', process_generate_options)
     app.add_config_value('autosummary_generate', [], True, [bool])
+    app.add_config_value('autosummary_generate_option', 'skip', False,
+                         ENUM('skip', 'warning', 'overwrite'))
```

Comment:
I need the better name for this configuration...

Reply:
I want to avoid autosummary_generate_option, because the term option associates the option of the directive.

my suggestions:

* autosummary_generate_behavior
* autosummary_generate_confirmation (too long ..)

Reply:
Thank you for comment. I reconsider this option again.

The `skip` will not help users. Because it hides conflicts. That causes the outputs are not updated. This is only added to keep compatibility. It is meaningless to me.

Next, the `warning` is a bit better than `skip`. It notices conflicts to users. Users can clear them by removing old stub files. I think the warning only helps who put user content as same name as stub file. Almost people feel harmful because they need to remove old stub files manually. So I think this option is useful only at first run.

The `overwrite` is a best way for almost case. Almost users will choose this option finally, I believe.

In conclusion, I think `autosummary_generate_overwrite = True | False` would be better than `autosummary_generate_option`. I withdraw `skip` option, and will be merged into 3.0 release because of breaking change.

What do you think?

Reply:
+1 for `autosummary_generate_overwrite`. 
Thanks for your summarizing!